src/rgw/rgw_acl_swift.cc

```diff
@@ -136,3 +140,124 @@ void RGWAccessControlPolicy_SWIFT::to_str(string& read, string& write)
   }
 }
 
+void RGWAccessControlPolicy_SWIFTAcct::add_grants(RGWRados * const store,
+                                                  const list<string>& uids,
+                                                  const int perm)
+{
+  for (const auto uid : uids) {
+    ACLGrant grant;
+    RGWUserInfo grant_user;
+
+    if (uid_is_public(uid)) {
+      grant.set_group(ACL_GROUP_ALL_USERS, perm);
+      acl.add_grant(&grant);
+    } else  {
+      rgw_user user(uid);
```

Comment:
rgw_user feels like it should be optimized more for cases like this, minor, but churns in all paths.